components/service/glean/src/main/java/mozilla/components/service/glean/storages/GenericScalarStorageEngine.kt

```diff
@@ -79,22 +85,23 @@ internal abstract class GenericScalarStorageEngine<ScalarType> : StorageEngine {
     )
 
     /**
-     * Deserialize the metrics with a lifetime = User that are on disk.
+     * Deserialize the metrics with a particular lifetime that are on disk.
      * This will be called the first time a metric is used or before a snapshot is
      * taken.
      *
-     * @return A [SharedPreferences] reference that will be used to initialize [userLifetimeStorage]
+     * @param lifetime a [Lifetime] to deserialize
+     * @return A [SharedPreferences] reference that will be used to initialize [pingLifetimeStorage]
+     *         or [userLifetimeStorage] or null if an invalid lifetime is used.
      */
     @Suppress("TooGenericExceptionCaught")
-    open fun deserializeUserLifetime(): SharedPreferences {
-        val prefs =
-            applicationContext.getSharedPreferences(this.javaClass.canonicalName, Context.MODE_PRIVATE)
+    open fun deserializeLifetime(lifetime: Lifetime): SharedPreferences? {
+        val prefs = getPrefsForLifetime(lifetime)
 
         val metrics = try {
-            prefs.all.entries
+            prefs!!.all.entries
```

Comment:
We're intentionally causing a `NullPointerException` here if `lifetime == Lifetime.Application`. However, `prefs` is a nullable and the else branch is really just for the `.all.entries` part. So what about we change this to `prefs?.all.entries` and add a comment there mentioning why are we using `?` in the first place?

Reply:
Fixed so I don't need to use either !! or ?.  It looks like `getSharedPreferences()` guarantees that it returns something, according to the docs.